### DISCUSSION TITLE
Chi-Square (χ²) and rrms Value

### DISCUSSION THREAD

**Discussion by RFauziahSM** [gimli-org/pyGIMLi#442] [Q&A]
Hi, everyone.
I'm still learning about pygimli, so please forgive my lack of knowledge about python.
I want to ask, in the following example https://www.pygimli.org/_examples_auto/3_dc_and_ip/plot_02_ert_field_data.html
what mathematical equation can I use to manually calculate the Chi-Square (χ²) and rrms value, so that it can produce the same value as Chi-Square (χ²) and rrms in the code? Thank You

**Comment by halbmy** [comment]
Obviously we need a tutorial for this. 

An (absolute) RMS (root mean square) error is defined by the root mean square of the misfit between data $d_i$ and model response $f_i(m)$:

$$
ARMS = \sqrt{\frac{1}{N} \sum_i^N \left(d_i-f_i(m)\right)^2}
$$

The result has the unit of the data, e.g. for traveltime the absolute error dominates. The unit could be s, ms, µs or ns depending on the method being used like global seismics, refraction seismics, ultrasonic or GPR, respectively.

In the relative RMS the misfit are weighted by the data points

$$
RRMS = \sqrt{\frac{1}{N} \sum_i^N \left(\frac{d_i-f_i(m)}{d_i}\right)^2}
$$

The RRMS value is typically multiplied by 100%. This is typical for ERT, where the dominating error is a relative one and it is the same for voltage, resistance or apparent resistivity.

If only "RMS" is given, one can deduce it from the unit. Sometimes no unit is given and RMS could also be the error-weighted misfit ($e_i$ are the error values)

$$
ERMS = \sqrt{\frac{1}{N} \sum_i^N \left(\frac{d_i-f_i(m)}{e_i}\right)^2}
$$

More specifically, one can specify the chi-square value

$$
\chi^2 = ERMS^2 = \frac{1}{N} \sum_i^N \left(\frac{d_i-f_i(m)}{e_i}\right)^2
$$

If a data transformation is used, one should take this into account. For ERT, one typically uses the logarithm of the apparent resistivity $d_i=\rho_i^a$. The absolute RMS of the logarithm

$$
ARMS = \sqrt{\frac{1}{N} \sum_i^N (\log d_i - \log f_i(m))^2} 
= \sqrt{\frac{1}{N} \sum_i^N (\log f_i(m)/d_i)^2}
$$

is the RMS of the ratio and thus related to the relative RMS.

**Comment by halbmy** [comment]
We recently extended one of the basic inversion tutorials with definition and discussion on those values:
https://www.pygimli.org/_tutorials_auto/3_inversion/plot_1-polyfit.html

**Reply by makeabhishek** [reply]
Dear @halbmy and @RFauziahSM
Sorry for jumping in the discussion.  I'm trying to implement traveltime inversion and looking to understand the mathematics behind the codes. 

Is it also applicable to traveltime inversion? When we perform traveltime inversion it give $\chi^2$, $\phi(m)=\phi_d(m)+\lambda \phi_m(m)$, where $\phi_d(m)$ is a data misfit measure and \phi_m(m) is model objective function or regularisation term. How does $\phi_d(m)$ and $\phi_m(m)$ calculated? 

What kind or regulariser does #pygimli use?

**Reply by halbmy** [reply]
$\Phi_d$ is defined in the last formula in the provided tutorial.

$\Phi_m$ depends on the chosen regularization that can be controlled. For mesh-based inversions, first order smoothness is the default, for others a damping (Marquardt method). For details see the regularization tutorial https://www.pygimli.org/_tutorials_auto/3_inversion/plot_5_Regularization.html
